### PR TITLE
Internal cleanup of TreehouseContent

### PR DESCRIPTION
**Pull request by swankjesse**
Rename TreehouseContent to TreehouseAppContent. The name TreehouseContent is already taken by Compose.

Rename RealBinding to ViewContentCodeBinding.

Move ViewContentCodeBinding into the same file as TreehouseAppContent.

Remove the unused type parameter on Content.

### CODE REVIEW

File: redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt [hunk 1]
```diff
@@ -84,7 +149,7 @@ internal class RealBinding<A : AppService>(
   private var treehouseUiOrNull: ZiplineTreehouseUi? = null
 
   /** Only accessed on [TreehouseDispatchers.ui]. */
-  private var firstDiff = true
+  private var diffCount = 0
```
Comment (JakeWharton) [inline redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt:152]: Why this change?

**Review by JakeWharton — APPROVED** [review]

File: redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt (hunk 1 above)
Comment (swankjesse) [reply redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt:152]: Yeah, this is off topic from the rest of the PR. I added `codeLoadCount` to replace tracking the previous binding, and I thought these two should be symmetric.

I’m contemplating about changing `CodeListener` to something less opinionated.

```
fun onCodeLoading(content: Content, view: TreehouseView, count: Int)
fun beforeDiffApplied(content: Content, view: TreehouseView, count: Int)
```

We don’t need this flexibility yet.
